src/Corax/Queries/MemoizationMatch.Provider.cs

```diff
@@ -6,9 +6,12 @@
 namespace Corax.Queries
 {
 
-    public unsafe struct MemoizationMatchProvider<TInner> : IMemoizationMatchSource
+    public unsafe class MemoizationMatchProvider<TInner> : IMemoizationMatchSource
```

Comment:
Why is this a class now? Is this instance shared? In what context? What about multi threading?

Reply:
Probably IDE changed it and I've not noticed it. I will revert that